src/network/networkd-address.c

```diff
@@ -809,8 +813,21 @@ int config_parse_label(
         if (r < 0)
                 return r;
 
-        if (!ifname_valid(rvalue)) {
-                log_syntax(unit, LOG_ERR, filename, line, 0, "Interface label is not valid or too long, ignoring assignment: %s", rvalue);
+        p = strchr(rvalue, ':');
+        if (p) {
+                r = safe_atou(p + 1, &k);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse label, ignoring: %s", rvalue);
+                        return 0;
+                }
+
+                strncpy(ifname, rvalue, p - rvalue);
```

Comment:
Buffer overflow here. There's nothing to guarantee that `p - rvalue <= sizeof ifname`.

Also, once you fix that, I think that `ifname` might not be NUL-terminated, so in the error message below `%.*s` should be used.